webapp/channels/src/components/sidebar/sidebar_header/sidebar_team_menu.tsx

```diff
@@ -0,0 +1,468 @@
+        );
+    }
+
+    let createTeamMenuItem: JSX.Element | null = null;
+    if (havePermissionToCreateTeam) {
+        createTeamMenuItem = (
+            <Menu.Item
+                disabled={isTeamsLimitReached}
+                leadingElement={(
+                    <PlusIcon
+                        size={18}
+                        aria-hidden='true'
+                    />
+                )}
+                onClick={onCreateTeamMenuItemClick}
+                labels={(
+                    <FormattedMessage
+                        id='sidebarLeft.teamMenu.createTeamMenuItem.primaryLabel'
+                        defaultMessage='Create a team'
+                    />
+                )}
+                trailingElements={isTeamCreateRestricted && <RestrictedIndicatorForCreateTeam isFreeTrial={isFreeTrial}/>}
+            />
+        );
+    }
+
+    const learnAboutTeamsMenuItem = (
+        <Menu.Item
+            className='learnAboutTeamsMenuItem'
+            onClick={onLearnAboutTeamsMenuItemClick}
+            leadingElement={(
+                <LightbulbOutlineIcon
+                    size={18}
+                    aria-hidden='true'
+                />
+            )}
+            labels={(
+                <FormattedMessage
+                    id='sidebarLeft.teamMenu.learnAboutTeamsMenuItem.primaryLabel'
+                    defaultMessage='Learn about teams'
+                />
+            )}
+        />
+    );
+
+    let pluginMenuItems: JSX.Element[] | null = null;
+    if (pluginInMainMenu.length > 0) {
+        pluginMenuItems = pluginInMainMenu.map((plugin) => {
+            function handleClick() {
+                if (plugin.action) {
+                    plugin.action();
+                }
+            }
+
+            return (
+                <Menu.Item
+                    id={`${plugin.id}_pluginmenuitem`}
+                    key={plugin.id}
+                    onClick={handleClick}
+                    labels={<span>{plugin.text}</span>}
+                />
+            );
+        });
+    }
+
+    return (
+        <Menu.Container
+            menuButton={{
+                id: 'sidebarTeamMenuButton',
+                class: 'btn btn-sm btn-quaternary btn-inverted',
+                children: (
+                    <>
+                        <span>{props.currentTeam.display_name}</span>
+                        <i className='icon icon-chevron-down'/>
+                    </>
+                ),
+            }}
+            menuButtonTooltip={{
+                text: tooltipText,
+            }}
+            menu={{
+                id: 'sidebarTeamMenu',
+                width: '225px',
+            }}
+        >
+            {invitePeopleMenuItem}
+            {addGroupsToTeamMenuItem}
+            {teamSettingsMenuItem}
+            {manageViewMembersMenuItem}
+            {manageGroupsMenuItem}
+            {leaveTeamMenuItem}
+            {Boolean(createTeamMenuItem) && <Menu.Separator/>}
```

Comment:
Should this also check `joinAnotherTeamMenuItem` as well?

Also, I'm wondering if we should implement a similar feature in this menu to the widget where the CSS for menu separators hides them if they're the first/last in the list or adjacent to one another one since that would avoid us needing to manage showing/hiding separators manually